Add troubleshooting docs (#1456)
alphagov/signon: docs/troubleshooting.md

# Troubleshooting

> This is a place to record workarounds to some common problems

Reviewer: ```suggestion
This is a place to record workarounds to some common problems.
```

## User didn't receive password reset email

> Signon current uses Amazon Simple Email Service (SES) to send emails to users. Amazon SES emails don’t pass strict DMARC setups. This means that some mail servers block signon email because they can't tell apart from phishing or spoof emails.

Reviewer: ```suggestion
Signon currently uses Amazon Simple Email Service (SES) to send emails to users. Amazon SES emails don’t pass strict DMARC setups. This means that some mail servers block Signon emails because they can't tell them apart from phishing or spoof emails.
```

> Signon currently uses Amazon Simple Email Service (SES) to send emails to users. Amazon SES emails don’t pass strict DMARC setups. This means that some mail servers block Signon emails because they can't tell them apart from phishing or spoof emails.

Reviewer: It's probably worth adding a link to either https://dmarc.org/ or https://en.wikipedia.org/wiki/DMARC.
```suggestion
Signon currently uses Amazon Simple Email Service (SES) to send emails to users. Amazon SES emails don’t pass strict [DMARC](https://dmarc.org/) setups. This means that some mail servers block Signon emails because they can't tell them apart from phishing or spoof emails.
```

> The proper fix to this is to move signon to use GOV.UK Notify, but until that happens we need to find a way to allow users to reset the passwords.

Reviewer: ```suggestion
The proper fix to this is to move Signon to use GOV.UK Notify, but until that happens we need to find a way to allow users to reset the passwords.
```

### Finding the reset password link

All the emails do is send users a link to reset their password. We can find this link and give it to them directly.

> First, open a Rails console (signon is on the backend box in Carrenza)

Reviewer: ```suggestion
First, open a Rails console (Signon is on the backend box in Carrenza).
```
